pkg/command/root.go

```diff
@@ -291,13 +331,22 @@ func newRootCmd() *cobra.Command {
 }
 
 // setLoggerVerbosity sets the verbosity of the logger if TANZU_CLI_LOG_LEVEL is set
-func setLoggerVerbosity() {
-	// Configure the log level if env variable TANZU_CLI_LOG_LEVEL is set
-	logLevel := os.Getenv(log.EnvTanzuCLILogLevel)
-	if logLevel != "" {
-		logValue, err := strconv.ParseInt(logLevel, 10, 32)
-		if err == nil {
-			log.SetVerbosity(int32(logValue))
+func setLoggerVerbosity(verbosity int) {
+	// If verbose global flag is passed set the log level verbosity  if not then check if env variable TANZU_CLI_LOG_LEVEL is set
+	if verbosity > 0 {
```

Comment:
I think this will always be true, since the default value for the flag is > 0.
Notice how the variable is still ignored:
```
$ TANZU_CLI_LOG_LEVEL=3 tz version
[i] I am level default
[i] I am level 1
version: v1.3.0-dev
buildDate: 2024-03-12
sha: f61583163-dirty
arch: arm64
```
Can you write a test for this?

Fixing this is a bit tricky.
If the user sets the `--verbose` flag then the flag value should be respected (ignoring `TANZU_CLI_LOG_LEVEL`); if the flag is not set, then `TANZU_CLI_LOG_LEVEL` should be honoured.
To know if the flag was set on the command-line we can:
1. set a default value that a user would not use like -1.  The problem is that the help text of the flag will show that default value.  That's not good.
2. don't set a default value (technically) by setting it to 0.  Cobra will not put it in the help text in that case.  Then if we see a value of 0, we assume the flag was not used.  BUT, do we want to allow the user to set `--verbose 0`?  It could make sense if they want to disable all logs maybe?  or else they cannot turn off level 1 logs.  So that approach is also not good.
3. ask Cobra if the flag was actually found on the command line.  We do this in `completeDeletePlugin()`. You can do it with `if !cmd.Flags().Changed("verbose") {verbose = -1}`

Reply:
Updated the if statement to `if verbosity >= 0 ` since we set the verbose to -1 if user doesn't set the flag